src/dioptra/restapi/v1/jobs/service.py

```diff
@@ -527,6 +525,157 @@ def get(
         return {"status": job.status, "id": job.resource_id}
 
 
+class JobIdMetricsService(object):
+    """The service methods for retrieving the metrics of a job by unique id."""
+
+    @inject
+    def __init__(
+        self,
+        job_id_mlflowrun_service: JobIdMlflowrunService,
+    ) -> None:
+        """Initialize the job metrics service.
+
+        All arguments are provided via dependency injection.
+
+        """
+        self._job_id_mlflowrun_service = job_id_mlflowrun_service
+
+    def get(
+        self,
+        job_id: int,
+        **kwargs,
+    ) -> list[dict[str, Any]]:
+        """Fetch a job's metrics by its unique id.
+
+        Args:
+            job_id: The unique id of the job.
+
+        Returns:
+            The metrics for the requested job if found, otherwise an error message.
+        """
+        from mlflow.tracking import MlflowClient
+
+        log: BoundLogger = kwargs.get("log", LOGGER.new())
+        log.debug("Get job metrics by id", job_id=job_id)
+
+        run_id: UUID = self._job_id_mlflowrun_service.get(job_id=job_id, **kwargs)[
```

Comment:
If the job is missing mlflowRunId, an error is raised. This can happen when a job errors out before attaching an mlflowRunId.

Recommend adding a check for this and returning an empty list of metrics instead of raising an error.